cmd/xmtpd-cli/commands/admin_setup.go

```diff
@@ -0,0 +1,195 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/viper"
+	"github.com/xmtp/xmtpd/pkg/blockchain"
+	"github.com/xmtp/xmtpd/pkg/config"
+	"go.uber.org/zap"
+)
+
+func setupAppChainAdmin(
+	ctx context.Context,
+	logger *zap.Logger,
+) (blockchain.IAppChainAdmin, error) {
+	var (
+		rpcURL     = viper.GetString("rpc-url")
+		configFile = viper.GetString("config-file")
+		privateKey = viper.GetString("private-key")
+	)
+	if rpcURL == "" {
+		return nil, fmt.Errorf("rpc-url is required")
+	}
+	if configFile == "" {
+		return nil, fmt.Errorf("config-file is required")
+	}
+	if privateKey == "" {
+		return nil, fmt.Errorf("private-key is required")
+	}
+
+	contracts, err := config.ContractOptionsFromEnv(configFile)
+	if err != nil {
+		return nil, fmt.Errorf("could not load config from file: %w", err)
+	}
+
+	client, err := blockchain.NewRPCClient(ctx, rpcURL)
+	if err != nil {
+		return nil, err
+	}
+	signer, err := blockchain.NewPrivateKeySigner(privateKey, contracts.AppChain.ChainID)
+	if err != nil {
+		return nil, fmt.Errorf("could not create signer: %w", err)
+	}
+
+	// NewAppChainAdmin expects a ParameterAdmin inside; this remains internal to the admin.
+	paramAdmin, err := blockchain.NewParameterAdmin(logger, client, signer, contracts)
+	if err != nil {
+		return nil, fmt.Errorf("could not create parameter admin: %w", err)
+	}
+
+	return blockchain.NewAppChainAdmin(logger, client, signer, contracts, paramAdmin)
+}
+
+func setupSettlementChainAdmin(
+	ctx context.Context,
+	logger *zap.Logger,
+) (blockchain.ISettlementChainAdmin, error) {
+	var (
+		rpcURL     = viper.GetString("rpc-url")
+		configFile = viper.GetString("config-file")
+		privateKey = viper.GetString("private-key")
+	)
+	if rpcURL == "" {
+		return nil, fmt.Errorf("rpc-url is required")
+	}
+	if configFile == "" {
+		return nil, fmt.Errorf("config-file is required")
+	}
+	if privateKey == "" {
+		return nil, fmt.Errorf("private-key is required")
+	}
+
+	contracts, err := config.ContractOptionsFromEnv(configFile)
+	if err != nil {
+		return nil, fmt.Errorf("could not load config from file: %w", err)
+	}
+
+	client, err := blockchain.NewRPCClient(ctx, rpcURL)
+	if err != nil {
+		return nil, err
+	}
+	signer, err := blockchain.NewPrivateKeySigner(privateKey, contracts.SettlementChain.ChainID)
+	if err != nil {
+		return nil, fmt.Errorf("could not create signer: %w", err)
+	}
+
+	paramAdmin, err := blockchain.NewParameterAdmin(logger, client, signer, contracts)
+	if err != nil {
+		return nil, fmt.Errorf("could not create parameter admin: %w", err)
+	}
+
+	return blockchain.NewSettlementChainAdmin(logger, client, signer, contracts, paramAdmin)
+}
+
+func setupNodeRegistryAdmin(
+	ctx context.Context,
+	logger *zap.Logger,
+) (blockchain.INodeRegistryAdmin, error) {
+	var (
+		privateKey = viper.GetString("private-key")
+		rpcURL     = viper.GetString("rpc-url")
+		configFile = viper.GetString("config-file")
```

Comment:
nit / future enhancement: we can come with an standard way of dealing with commonly used flags, to simplify the command creation process.
Something like `GetPrivateKeyFlag`, `GetRPCURLFlag`, etc. that calls viper under the hood, the keys are constants, etc